### PR TITLE
Add migrating go video.

### PR DESCRIPTION
**Pull request by amouat**
## Type of change

Add the "Migrating a Dockerfile for a Go application to use Chainguard Images" video.

### How should this PR be tested?

Please check it looks ok, has the correct embed and no obvious typos

### CODE REVIEW

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+Let's check it works.
+
+Okay, so that works identically.
+
+A one-line change, and we've made no difference to the actual running application.
+
+But let's investigate this a little bit more.
+
+So if I do 'docker images' on go-web-app, well this is a pretty large image at 892 megabytes.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:92]:
```suggestion
So if I do `docker images` on `go-web-app`, well this is a pretty large image at 892 megabytes.
```

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+And I'm going to use the Chainguard static image to house the final production application.
+
+So the static image is really simple.
+
+It has very little in it, just sort of the minimum you need to run a typical Linux application.
+
+So it has TLS certificates so you can talk to web applications over TLS.
+
+It has Unix directories like /tmp and /home, and it doesn't have a lot else.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:130]:
```suggestion
It has Unix directories like `/tmp` and `/home`, and it doesn't have a lot else.
```

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+It's only a few megabytes in size, but contains a few more things than say scratch that you need for typical applications.
+
+OK, what we're going to do here is I'm going to copy from the previous build to the Dockerfile up here.
+
+I'm going to copy the hello executable, which is at /work hello to /hello.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:136]:
```suggestion
I'm going to copy the `hello` executable, which is at `/work/hello` to `/hello`.
```

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+Now it said copy from builder, so I'll need to name this first step builder.
+
+And that will build this hello executable and copy it into my production image.
+
+And then my entry point is now /hello.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:142]:
```suggestion
And then my entry point is now `/hello`.
```

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+So there is one more thing we need to do.
+
+The static image does not include any libraries like glibc or anything.
+
+It just has it's just for running statically compiled binaries.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:148]: I think something is missing in this line

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+So we need to tell Go to produce a statically compiled binary that contains everything it needs to run and doesn't rely on system libraries.
+
+And we can do that by saying cgo enabled equals zero.
+
+In some cases, you might find you need to pass a few more flags, depending on which Go libraries you use.
+
+And I'll link a blog in the notes that explains when you need to do this and what you need to do.
+
+But in this case, cgo enabled equals zero will allow us to build our static binary.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:158]:
```suggestion
But in this case, `CGO_enabled=0` will allow us to build our static binary.
```

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+I think I still have the old one running.
+
+So let's get rid of the old web app.
+
+No, I always get that wrong.
+
+OK, and let's run this one.
+
+I call it web-app-distroless.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:188]:
```suggestion
I call it `web-app-distroless`.
```

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+
+That's running.
+
+Let's see if it works.
+
+Excellent.
+
+So this application still works exactly the same as it did at the start of this video.
+
+But if I take a look at the web app distroless, the big difference is that now it's only eight point five megabytes in size.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:200]:
```suggestion
But if I take a look at the web app distroless, the big difference is that now it's only 8.5 megabytes in size.
```

File: content/chainguard/chainguard-images/videos/migrating_go.md [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+
+So we went from the original Golang image, which is eight hundred and ninety two megabytes, to a Chainguard Image, which was seven hundred seventy five megabytes, down to eight point five one megabytes.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/migrating_go.md:202]:
```suggestion
So we went from the original Golang image, which is 892 megabytes, to a Chainguard Image, which was 775 megabytes, down to 8.51 megabytes.
```

**Review by ltagliaferri — APPROVED** [review]
Some readability changes in the transcript

File: content/chainguard/chainguard-images/videos/migrating_go.md (hunk 5 above)
Comment (amouat) [reply content/chainguard/chainguard-images/videos/migrating_go.md]: Should be fixed.

**Comment by amouat** [comment]
> Some readability changes in the transcript

Thanks! That was a fair bit of work :)

**Pull request merged**